### DISCUSSION TITLE
Incremental Builds without Gatsby Cloud (is it possible?)

### DISCUSSION THREAD

**Discussion by Shnass** [gatsbyjs/gatsby#30368] [Help]
Hello. I am novice to webdevelopment and to Gatsby aswell, have chosen it for my project almost randomly. I am doing quite a small site where I have strapi as backend. Both Strapi and Gatsby are located on the same server (ubuntu), and I use Strapi lifecycles to initiate gatsby rebuilds, but each time I update anything (like name of the post or picture) it takes some seconds (let's say 20-30) to rebuild my frontend, which I find quite a significant time. I have discovered this article https://www.gatsbyjs.com/blog/2020-04-22-announcing-incremental-builds/ and I wonder if there any way to run Incremental Builds without using gatsby cloud?

At this moment I fell like I'm misunderstanding the whole concept of Gatsby, or that I'm misusing it somehow

**Comment by herecydev** [comment]
Yep it should work out the box now, see here: https://www.gatsbyjs.com/docs/reference/release-notes/v3.0/#incremental-builds-in-oss

**Comment by Shnass** [comment]
in the presentation video it took him 30 seconds to rebuild project just to add two words to the description of one item, I still have the feeling that I'm misunderstanding something

**Reply by herecydev** [reply]
That sounds about right, it's unfortunately not as sophisticated as gatsby cloud is. You're effectively spinning up a lot of gatsby for it to realise that very little has changed, however if you're building 10'000 pages and your build is taking a long time, this will greatly reduce the time for changes to be deployed. You only pay the cost for pages/data which has changed.

I guess 30 seconds might still be unreasonable for you, ultimately that's the tradeoff between the free DIY version of gatsby cloud.
